// @demox-labs/aleo-wallet-adapter-base/dist/errors.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WalletTransactionError = exports.WalletRecordsError = exports.WalletDecryptionError = exports.WalletDecryptionNotAllowedError = exports.WalletWindowClosedError = exports.WalletWindowBlockedError = exports.WalletTimeoutError = exports.WalletSignTransactionError = exports.WalletSignMessageError = exports.WalletSendTransactionError = exports.WalletNotConnectedError = exports.WalletKeypairError = exports.WalletPublicKeyError = exports.WalletAccountError = exports.WalletDisconnectionError = exports.WalletDisconnectedError = exports.WalletNotSelectedError = exports.WalletConnectionError = exports.WalletConfigError = exports.WalletLoadError = exports.WalletNotReadyError = exports.WalletError = void 0;
class WalletError extends Error {
    error;
    // eslint-disable-next-line @typescript-eslint/explicit-module-boundary-types
    constructor(message, error) {
        super(message);
        this.error = error;
    }
}
exports.WalletError = WalletError;
class WalletNotReadyError extends WalletError {
    name = 'WalletNotReadyError';
}
exports.WalletNotReadyError = WalletNotReadyError;
class WalletLoadError extends WalletError {
    name = 'WalletLoadError';
}
exports.WalletLoadError = WalletLoadError;
class WalletConfigError extends WalletError {
    name = 'WalletConfigError';
}
exports.WalletConfigError = WalletConfigError;
class WalletConnectionError extends WalletError {
    name = 'WalletConnectionError';
}
exports.WalletConnectionError = WalletConnectionError;
class WalletNotSelectedError extends WalletError {
    name = 'WalletNotSelectedError';
}
exports.WalletNotSelectedError = WalletNotSelectedError;
class WalletDisconnectedError extends WalletError {
    name = 'WalletDisconnectedError';
}
exports.WalletDisconnectedError = WalletDisconnectedError;
class WalletDisconnectionError extends WalletError {
    name = 'WalletDisconnectionError';
}
exports.WalletDisconnectionError = WalletDisconnectionError;
class WalletAccountError extends WalletError {
    name = 'WalletAccountError';
}
exports.WalletAccountError = WalletAccountError;
class WalletPublicKeyError extends WalletError {
    name = 'WalletPublicKeyError';
}
exports.WalletPublicKeyError = WalletPublicKeyError;
class WalletKeypairError extends WalletError {
    name = 'WalletKeypairError';
}
exports.WalletKeypairError = WalletKeypairError;
class WalletNotConnectedError extends WalletError {
    name = 'WalletNotConnectedError';
}
exports.WalletNotConnectedError = WalletNotConnectedError;
class WalletSendTransactionError extends WalletError {
    name = 'WalletSendTransactionError';
}
exports.WalletSendTransactionError = WalletSendTransactionError;
class WalletSignMessageError extends WalletError {
    name = 'WalletSignMessageError';
}
exports.WalletSignMessageError = WalletSignMessageError;
class WalletSignTransactionError extends WalletError {
    name = 'WalletSignTransactionError';
}
exports.WalletSignTransactionError = WalletSignTransactionError;
class WalletTimeoutError extends WalletError {
    name = 'WalletTimeoutError';
}
exports.WalletTimeoutError = WalletTimeoutError;
class WalletWindowBlockedError extends WalletError {
    name = 'WalletWindowBlockedError';
}
exports.WalletWindowBlockedError = WalletWindowBlockedError;
class WalletWindowClosedError extends WalletError {
    name = 'WalletWindowClosedError';
}
exports.WalletWindowClosedError = WalletWindowClosedError;
class WalletDecryptionNotAllowedError extends WalletError {
    name = 'WalletDecryptionNotAllowedError';
}
exports.WalletDecryptionNotAllowedError = WalletDecryptionNotAllowedError;
class WalletDecryptionError extends WalletError {
    name = 'WalletDecryptionError';
}
exports.WalletDecryptionError = WalletDecryptionError;
class WalletRecordsError extends WalletError {
    name = 'WalletRecordsError';
}
exports.WalletRecordsError = WalletRecordsError;
class WalletTransactionError extends WalletError {
    name = 'WalletTransactionError';
}
exports.WalletTransactionError = WalletTransactionError;
